### PR TITLE
Small refactor of the library

### PR DESCRIPTION
**Pull request by foxis**
* refactor of the code
* multiple networks
* reconnect on lost wifi
* support open networks
* GPL v3.0 license

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
+
+This library tries to connect to configured APs first. Then if those fail ,tries to scan for open networks and connect to those.
+If it is not possible to connect to neither configured nor open networks - it falls back to AP mode.
+
 # Installation
 
 ## Install as ZIP
 
-* [Click here to download the ZIP file](https://github.com/jeroenvermeulen/JeVe_EasyOTA/archive/master.zip)
+* [Click here to download the ZIP file](https://github.com/foxis/EasyOTA/archive/master.zip)
```
Comment (jeroenvermeulen) [inline README.md:21]: This change should not be in the PR :-)

File: README.md [hunk 2]
```diff
@@ -18,38 +27,44 @@ Library which makes it easy to add support for Over-The-Air (OTA) updates to you
 
 * Open a terminal window
 * Go to your Arduino libraries dir: `cd ~/Documents/Arduino/libraries`
-* Execute: `git clone https://github.com/jeroenvermeulen/JeVe_EasyOTA.git`
+* Execute: `git clone https://github.com/foxis/EasyOTA.git`
```
Comment (jeroenvermeulen) [inline README.md:30]: This change should not be in the PR :-)

File: examples/OTA/OTA.ino [hunk 3]
```diff
@@ -2,20 +2,20 @@
  * To start mDNS monitor (OSX) exec:   dns-sd -B _arduino._tcp
  */
 
-#include <JeVe_EasyOTA.h>  // https://github.com/jeroenvermeulen/JeVe_EasyOTA/blob/master/JeVe_EasyOTA.h
+#include <EasyOTA.h>  // https://github.com/foxis/EasyOTA/blob/master/EasyOTA.h
```
Comment (jeroenvermeulen) [inline examples/OTA/OTA.ino:5]: Please keep my Github URL (for now)

File: examples/OTA_LCD_I2C/OTA_LCD_I2C.ino [hunk 4]
```diff
@@ -5,13 +5,13 @@
  * To start mDNS monitor (OSX) exec:   dns-sd -B _arduino._tcp
  */
 
-#include <JeVe_EasyOTA.h>           // https://github.com/jeroenvermeulen/JeVe_EasyOTA/blob/master/src/JeVe_EasyOTA.h
+#include <EasyOTA.h>           // https://github.com/foxis/EasyOTA/blob/master/src/EasyOTA.h
```
Comment (jeroenvermeulen) [inline examples/OTA_LCD_I2C/OTA_LCD_I2C.ino:8]: Same here

File: examples/OTA_Oled_U8g2/OTA_Oled_U8g2.ino [hunk 5]
```diff
@@ -2,7 +2,7 @@
  * To start mDNS monitor (OSX) exec:   dns-sd -B _arduino._tcp
  */
 
-#include <JeVe_EasyOTA.h>  // https://github.com/jeroenvermeulen/JeVe_EasyOTA/blob/master/src/JeVe_EasyOTA.h
+#include <EasyOTA.h>  // https://github.com/foxis/EasyOTA/blob/master/src/EasyOTA.h
```
Comment (jeroenvermeulen) [inline examples/OTA_Oled_U8g2/OTA_Oled_U8g2.ino:5]: Same here

File: library.json [hunk 6]
```diff
@@ -1,23 +1,24 @@
 {
-  "name": "JeVe_EasyOTA",
+  "name": "EasyOTA",
   "authors":
   {
-    "name": "Jeroen Vermeulen",
-    "email": "info@jeve.eu",
-    "url": "https://www.jeroenvermeulen.eu/"
+    "name": "Andrius Mikonis",
```
Comment (jeroenvermeulen) [inline library.json:5]: Could we be both author?

File: library.json [hunk 7]
```diff
@@ -1,23 +1,24 @@
...
+    "email": "andrius.mikonis@gmail.com",
+    "url": ""
   },
   "keywords": "OTA, Over The Air Updates, ArduinoOTA",
-  "description": "Library which makes it easy to add support for Over-The-Air (OTA) updates to your project. Works with ESP8266 and ESP32.",
+  "description": "Library which makes it easy to add support for Over-The-Air (OTA) updates to your project. Works with ESP8266 and ESP32. A refactored version of JeVe_EasyOTA with added multiple network support and easier interface.",
```
Comment (jeroenvermeulen) [inline library.json:10]: Let's make it into one lib!

File: library.properties [hunk 8]
```diff
@@ -1,9 +1,9 @@
-name=JeVe_EasyOTA
-version=2.2.0
-author=Jeroen Vermeulen <info@jeve.eu>
-maintainer=Jeroen Vermeulen <info@jeve.eu>
+name=EasyOTA
+version=3.0.0
+author=Andrius Mikonis <andrius.mikonis@gmail.com>
+maintainer=Andrius Mikonis <andrius.mikonis@gmail.com>
```
Comment (jeroenvermeulen) [inline library.properties:4]: Same here

**Comment by jeroenvermeulen** [comment]
Too bad you renamed the main files, hard to see what you changed...

File: library.properties (hunk 8 above)
Comment (foxis) [reply library.properties:4]: Adding to authors to library.json was easy, but I don't quite know how to do it with library.property file... maybe add them separated by comma ?
I'll update the PR if it's ok.

**Comment by foxis** [comment]
I think you can reject this PR and look at the other one.
Since I am using my version in several projects it would be a pain to rename(especially temporarily) included headers... So all the code merging will be done in a separate branch on my fork.
I'll merge it into my master once we iron out all the bits and pieces...

**Pull request merged**